src/app/pages/data-element/data-element.component.ts

```diff
@@ -247,4 +253,16 @@ export class DataElementComponent implements OnInit {
         .filter((section) => section.fields.length > 0),
     };
   }
+
+  private subscribeDataRequestChanges() {
+    this.broadcast
+      .on('data-request-added')
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe(() => {
```

Comment:
Should this be a map or switch map to avoid having nesting subscriptions?

Reply:
Is that not only the case if we want to cancel the first subscription on the trigger? I think we want to stay subscribed so other events also trigger this method? Unless I'm being dense?

Reply:
I believe your latest code solves this, thanks!